cli/src/action/circuit/builder.rs

```diff
@@ -177,13 +177,25 @@ impl CreateCircuitMessageBuilder {
             }
         };
 
-        let node = make_splinter_node(&node_id, &endpoint)?;
-        self.nodes.push(node);
-        Ok(())
+        self.add_node(&node_id, &endpoint)
     }
 
     pub fn add_node(&mut self, node_id: &str, node_endpoint: &str) -> Result<(), CliError> {
         let node = make_splinter_node(node_id, node_endpoint)?;
+
+        if self.nodes.iter().any(|node| node.node_id == node_id) {
+            return Err(CliError::ActionError(format!(
+                "Duplicate node ID detected: {}",
+                node_id
+            )));
+        }
+        if self.nodes.iter().any(|node| node.endpoint == node_endpoint) {
+            return Err(CliError::ActionError(format!(
+                "Duplicate node endpoint detected: {}",
+                node_endpoint
+            )));
+        }
```

Comment:
It's not a huge difference, buy you could probably refactor this to require a single iteration over nodes?

Reply:
Updated